src/main/java/com/ays/backend/user/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package com.ays.backend.user.service.impl;
+
+import com.ays.backend.user.exception.UserNotFoundException;
+import com.ays.backend.user.model.entities.User;
+import com.ays.backend.user.repository.UserRepository;
+import com.ays.backend.user.security.JwtUserDetails;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
```

Comment:
We can make the implementation classes package-private so that they can only be accessed through interfaces.